File: Problem Solver/src/Problem016.cpp
#include "Problem016.h"
#include "EulerHelper.h"
#include "InfInt.h"
#include <string>

using namespace std;

Problem016::Problem016()
{
    //ctor
}

Problem016::~Problem016()
{
    //dtor
}

void Problem016::Solve(){
    InfInt x = 2;
    for (int i = 2; i <= 1000; i++){
        x = x * 2;
    }

    string ans = EulerHelper::sumOfDigits(x.toString()).toString();

    cout << ans << endl;
}

File: ProblemSolver/ProblemSolver/src/Problem016.cs
/*
Power digit sum
Problem 16
215 = 32768 and the sum of its digits is 3 + 2 + 7 + 6 + 8 = 26.

What is the sum of the digits of the number 21000?
*/
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Numerics;

namespace ProblemSolver.src
{
    class Problem016 : SolvedProblems
    {
        public override void Solve()
        {
            BigInteger x = BigInteger.Pow(2, 1000);
            EulerHelper.printAns(EulerHelper.sumDigits(x.ToString()).ToString());
        }
    }
}
